### PR TITLE
super digit in java

### LINKED ISSUE #2242: Calculate Super digit of a number
# Aim

-Calculate Super digit of a number

# Details

- Calculate Super digit of a number using basic arithmetic algorithm 

### Programming language

- [x] C
- [ ] C++
- [ ] Java
- [ ] Python

### PR DESCRIPTION
**Pull request by Rounak-chauhan**
super digit of number in java

Fixes #2242 

Replace #2242  with the issue number which is fixed in this PR

## Type of change

_Please delete options that are not relevant._

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation Update

# Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have created a helpful and easy to understand `README.md`
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests/screenshots(if any) that prove my fix is effective or that my feature works

### CODE REVIEW

File: Hackerank/superdigit.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+public class superdigit {
```
Comment (sameeksharl) [inline Hackerank/superdigit.java:1]: @Rounak-chauhan add question here

File: Hackerank/superdigit.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    		Integer digitSum = 0;    		
+    		for (char num : x.toString().toCharArray()) {       			
+				digitSum += Character.getNumericValue(num);
+			}
+    		return getSuperDigit(digitSum);
+    	}
+	}
+
+    
+}
```
Comment (sameeksharl) [inline Hackerank/superdigit.java:22]: add test cases here

**Comment by Rounak-chauhan** [comment]
i had done the required changes please merge it now if it is ok

**Comment by Amit366** [comment]
@Rounak-chauhan add it to java file in hackerrank

**Comment by Amit366** [comment]
@Rounak-chauhan this is same as https://gist.github.com/FelixCoutinho/13c0fb8b230c7bb39bc533dd4e5a94a1

**Pull request closed without merging**